packages/nextjs/components/Modal.tsx

```diff
@@ -0,0 +1,44 @@
+/**
+ * Modal component: A reusable modal component.
+ */
+type ModalProps = {
+  trigger: JSX.Element;
+  modalTitle: string;
+  modalContent: JSX.Element;
+  action?: {
```

Comment:
It can be two or more buttons, for example "cancel" and "save". So probably we need to include this in content too

Reply:
Exactly, the actions can be included in the modal content.

In the exampe-ui the send is in the modalContent.

The action is optional cause I had a specific use for passing it in.

It can also be removed if there is no need to leave the option there.